### PR TITLE
Migration extensions - UI fixes and vBump

### PR DESCRIPTION
**Pull request by aasimkhan30**
Added more details in migration status dialog
Before: 
image
After:
image

Fixed Migration Cutover Dialog to match the mockups
Before:
image

After:
image

Fixed DMS selection page and made UI improvements:
Before:
image
After:
image

### CODE REVIEW

File: extensions/sql-migration/src/api/utils.ts [hunk 1]
```diff
@@ -40,3 +40,22 @@ export function getSqlServerName(majorVersion: number): string | undefined {
...
 	}
 }
+
+export function convertTimeToDuration(time: number): string {
+	let seconds = (time / 1000).toFixed(1);
+	let minutes = (time / (1000 * 60)).toFixed(1);
+	let hours = (time / (1000 * 60 * 60)).toFixed(1);
+	let days = (time / (1000 * 60 * 60 * 24)).toFixed(1);
+	if (time / 1000 < 60) {
+		return seconds + ' Sec';
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/api/utils.ts:50]: These values should be localized

File: extensions/sql-migration/src/api/utils.ts [hunk 2]
```diff
@@ -40,3 +40,22 @@ export function getSqlServerName(majorVersion: number): string | undefined {
 			return undefined;
 	}
 }
+
+export function convertTimeToDuration(time: number): string {
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/api/utils.ts:44]: Would suggest adding a comment explaining what a "duration" is (with examples)

File: extensions/sql-migration/src/dialog/createSqlMigrationService/createSqlMigrationServiceDialog.ts [hunk 3]
```diff
@@ -492,4 +546,10 @@ export class CreateSqlMigrationServiceDialog {
 			level: level
 		};
 	}
+
+	private enableForm(enable: boolean): void {
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/dialog/createSqlMigrationService/createSqlMigrationServiceDialog.ts:550]: `setFormEnabledState` or something - `enableForm` implies that this function is used for specifically enabling the form which isn't necessarily true.

File: extensions/sql-migration/src/wizard/integrationRuntimePage.ts [hunk 4]
```diff
@@ -104,69 +117,323 @@ export class IntergrationRuntimePage extends MigrationWizardPage {
...
+			}
+		}).component();
+
+
+		this._copy1 = this._view.modelBuilder.button().withProps({
+			iconPath: IconPathHelper.copy,
+		}).component();
+
+		this._copy1.onDidClick((e) => {
+			vscode.env.clipboard.writeText(<string>this._authKeyTable.dataValues![0][1].value);
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/integrationRuntimePage.ts:285]: You should await this

File: extensions/sql-migration/src/wizard/integrationRuntimePage.ts [hunk 5]
```diff
@@ -104,69 +117,323 @@ export class IntergrationRuntimePage extends MigrationWizardPage {
...
+			vscode.window.showInformationMessage(constants.SERVICE_KEY_COPIED_HELP);
+		});
+
+		this._copy2 = this._view.modelBuilder.button().withProps({
+			iconPath: IconPathHelper.copy
+		}).component();
+
+		this._copy2.onDidClick((e) => {
+			vscode.env.clipboard.writeText(<string>this._authKeyTable.dataValues![1][1].value);
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/integrationRuntimePage.ts:294]: await

File: extensions/sql-migration/src/wizard/integrationRuntimePage.ts [hunk 6]
```diff
@@ -104,69 +117,323 @@ export class IntergrationRuntimePage extends MigrationWizardPage {
...
+			let index = -1;
+			if (this.migrationStateModel._sqlMigrationService) {
+				index = (<azdata.CategoryValue[]>this._dmsDropdown.values).findIndex(v => v.displayName.toLowerCase() === this.migrationStateModel._sqlMigrationService.name.toLowerCase());
+			}
+			if (index !== -1) {
+				this._dmsDropdown.value = this._dmsDropdown.values[index];
+			} else {
+				this._dmsDropdown.value = this._dmsDropdown.values[0];
+			}
+		} catch (e) {
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/integrationRuntimePage.ts:415]: You don't want to at least log something to console?

File: extensions/sql-migration/src/wizard/integrationRuntimePage.ts [hunk 7]
```diff
@@ -104,69 +117,323 @@ export class IntergrationRuntimePage extends MigrationWizardPage {
...
+			this._dmsDropdown.loading = false;
 		}
 
 	}
 
 	private async loadMigrationServiceStatus(): Promise<void> {
 		this._statusLoadingComponent.loading = true;
 		try {
-			this._migrationDetailsContainer.clearItems();
+			this.loadStatus();
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/integrationRuntimePage.ts:426]: await

File: extensions/sql-migration/src/wizard/integrationRuntimePage.ts [hunk 8]
```diff
@@ -192,228 +459,61 @@ export class IntergrationRuntimePage extends MigrationWizardPage {
...
-						});
-					} else {
-						connectionStatus.updateProperties(<azdata.InfoBoxComponentProperties>{
-							text: constants.SERVICE_NOT_READY(this.migrationStateModel._sqlMigrationService!.name),
-							style: 'error'
-						});
-					}
-
-					connectionStatusLoader.loading = false;
+				this.migrationStateModel._nodeNames = migrationServiceMonitoringStatus.nodes.map((node) => {
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/integrationRuntimePage.ts:462]: FYI you can shorten this to `.map(node => node.nodeName);` - no need for all that extra stuff.

File: extensions/sql-migration/src/wizard/integrationRuntimePage.ts [hunk 9]
```diff
@@ -192,228 +459,61 @@ export class IntergrationRuntimePage extends MigrationWizardPage {
...
-							text: constants.SERVICE_READY(this.migrationStateModel._sqlMigrationService!.name, this.migrationStateModel._nodeNames.join(', ')),
-							style: 'success'
-						});
-					} else {
-						connectionStatus.updateProperties(<azdata.InfoBoxComponentProperties>{
-							text: constants.SERVICE_NOT_READY(this.migrationStateModel._sqlMigrationService!.name),
-							style: 'error'
-						});
-					}
+				if (state === 'Online') {
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/integrationRuntimePage.ts:467]: This value should probably be a constant somewhere

File: extensions/sql-migration/src/wizard/integrationRuntimePage.ts [hunk 10]
```diff
@@ -192,228 +459,61 @@ export class IntergrationRuntimePage extends MigrationWizardPage {
...
-					]
-				);
+				this._authKeyTable.updateProperties({
+					dataValues: data
+				});
 			}
-		} catch (error) {
-			console.log(error);
-			this._migrationDetailsContainer.clearItems();
+		} catch (e) {
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/integrationRuntimePage.ts:512]: Not logging error to console or anything?

**Review by kburtram — APPROVED** [review]
LGTM
